server/utils/assessments/suitabilityAssessmentAdjacentPage.test.ts

```diff
@@ -1,247 +1,229 @@
 import { retrieveOptionalQuestionResponseFromFormArtifact } from '../retrieveQuestionResponseFromFormArtifact'
-import { suitabilityAssessmentAdjacentPage } from './suitabilityAssessmentAdjacentPage'
+import { SuitabilityAssessmentPageName, suitabilityAssessmentAdjacentPage } from './suitabilityAssessmentAdjacentPage'
+import { startDateOutsideOfNationalStandardsTimescales } from '../applications/startDateOutsideOfNationalStandardsTimescales'
+import Rfap from '../../form-pages/apply/risk-and-need-factors/further-considerations/rfap'
+import SelectApType from '../../form-pages/apply/reasons-for-placement/type-of-ap/apType'
+import { ApprovedPremisesAssessment } from '../../@types/shared'
 
 jest.mock('../retrieveQuestionResponseFromFormArtifact')
 jest.mock('../applications/shouldShowContingencyPlanPages')
-jest.mock('../applications/noticeTypeFromApplication')
+jest.mock('../applications/startDateOutsideOfNationalStandardsTimescales')
+
+const mockApplicationOfType = (apType: string, assessment: ApprovedPremisesAssessment) => {
+  switch (apType) {
+    case 'rfap': {
+      when(retrieveOptionalQuestionResponseFromFormArtifact)
+        .calledWith(assessment.application, SelectApType, 'type')
+        .mockReturnValue('')
+
+      when(retrieveOptionalQuestionResponseFromFormArtifact)
+        .calledWith(assessment.application, Rfap, 'needARfap')
+        .mockReturnValue('yes')
+
+      break
+    }
+    default: {
+      when(retrieveOptionalQuestionResponseFromFormArtifact)
+        .calledWith(assessment.application, Rfap, 'needARfap')
+        .mockReturnValue('')
+
+      when(retrieveOptionalQuestionResponseFromFormArtifact)
+        .calledWith(assessment.application, SelectApType, 'type')
+        .mockReturnValue(apType)
+    }
+  }
+}
 
 describe('suitabilityAssessmentAdjacentPage', () => {
-  ;(
-    [
-      {
-        name: 'rfap',
-        pageName: 'rfap-suitability',
-        applicationQuestion: { name: 'needARfap', answer: 'yes' },
-      },
-      {
-        name: 'esap',
-        pageName: 'esap-suitability',
-        applicationQuestion: { name: 'type', answer: 'esap' },
-      },
-      {
-        name: 'pipe',
-        pageName: 'pipe-suitability',
-        applicationQuestion: { name: 'type', answer: 'pipe' },
-      },
-    ] as const
-  ).forEach(apType => {
+  const assessment = assessmentFactory.build()
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  describe('With the suitability-assessment page', () => {
+    describe.each(['rfap', 'esap', 'pipe'])('with an %s application', apType => {
+      beforeEach(() => {
+        mockApplicationOfType(apType, assessment)
+      })
+
+      it('should return the correct next page', () => {
+        expect(suitabilityAssessmentAdjacentPage(assessment, 'suitability-assessment')).toEqual(`${apType}-suitability`)
+      })
+    })
+  })
+
+  describe.each(['rfap', 'esap', 'pipe'])('with an %s application', apType => {
```

Comment:
If we cast the `['rfap', 'esap', 'pipe']` array `as const` then on line 65 we can type the `pageName` instead of casting it. It might event be useful to save the array to a const and share it between the tests that are using it.

Reply:
Good call! Have done that